src/stirling/source_connectors/socket_tracer/uprobe_manager.cc

```diff
@@ -280,53 +281,78 @@ StatusOr<std::vector<std::filesystem::path>> FindHostPathForPIDLibs(
                                 HostPathForPIDPathSearchType::kSearchTypeEndsWith);
 }
 
+// ssl_lib_matches allows customizing the search of shared object files
+// that need to be traced with the SSL_write and SSL_read uprobes.
+// In dynamically linked cases, it's likely that there are two
+// shared libraries (libssl and libcrypto). While statically linked
+// cases are contained within the same binary. ssl_library_match
+// allows for handling both cases. If there are two libraries present
+// in the std::vector, the former is assumed to be libssl and the
+// latter libcrypto.
+using ssl_lib_matcher =
+    std::vector<std::tuple<std::vector<std::string_view>, HostPathForPIDPathSearchType>>;
+
+ssl_lib_matcher kLibSSLMatchers = {
+    {
+        {"libssl.so.1.1", "libcrypto.so.1.1"},
+        HostPathForPIDPathSearchType::kSearchTypeEndsWith,
+    },
+    {
+        {kLibNettyTcnativePrefix},
+        HostPathForPIDPathSearchType::kSearchTypeContains,
+    },
+};
+
 // Return error if something unexpected occurs.
 // Return 0 if nothing unexpected, but there is nothing to deploy (e.g. no OpenSSL detected).
 StatusOr<int> UProbeManager::AttachOpenSSLUProbesOnDynamicLib(uint32_t pid) {
-  constexpr std::string_view kLibSSL = "libssl.so.1.1";
-  constexpr std::string_view kLibCrypto = "libcrypto.so.1.1";
-  const std::vector<std::string_view> lib_names = {kLibSSL, kLibCrypto};
+  for (auto ssl_library_match : kLibSSLMatchers) {
+    const std::vector<std::string_view> lib_names = std::get<0>(ssl_library_match);
+    auto const search_type = std::get<1>(ssl_library_match);
```

Comment:
nit: `const auto` instead of `auto const`

Reply:
Updated.